cmake/leatherman.cmake

```diff
@@ -147,3 +167,19 @@ function(get_commit_string varname)
         set(${varname} " (commit ${GIT_SHA1})" PARENT_SCOPE)
     endif()
 endfunction()
+
+include(GenerateExportHeader)
+# Usage: symbol_exports(TARGET HEADER)
+#
+# Generate the export header for restricting symbols exported from the library,
+# and configure the compiler. Restricting symbols has several advantages, noted
+# at https://gcc.gnu.org/wiki/Visibility.
+macro(symbol_exports target header)
+    generate_export_header(${target} EXPORT_FILE_NAME "${header}")
+    if (NOT APPLE)
```

Comment:
AIX will safely ignore it. Apple will try to use it, but then fail to export things we need so the libraries are unusable. There were some hangups we haven't figured out.

This worked at https://github.com/puppetlabs/cpp-pcp-client/blob/master/lib/CMakeLists.txt#L41-L50, so it should be safe for all our current platforms.